Creating TravellingProblem.Java

PedramSadr opened the pull request:
Ramin,
Can you please review this code?

PedramSadr committed:
Creating TravellingProblem.Java

rfarhanian commented on cs-clrs/src/test/java/com/mmnaseri/cs/clrs/ch25/TrvellingProblem.java:
```diff
@@ -0,0 +1,34 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class TrvellingProblem extends BaseAllPairsShortestPathFinderTest{
```
Please create the following cities in your graph:

London, Paris, Asalooye, Tehran, Sabzevaar, Kish, Khaash

Following are the price of the tickets between cities: 
London ---> Paris (250 Pound)
London --> Tehran (600 Pound)
Paris --> Tehran (550 Pound)
Paris--> Asalooyeh (100 Pound)
Tehran --> Sabzevaar (20 Pound)
Asalooyeh --> Tehran (20 Pound)
Asalooyeh --> Kish (99 Pound)
Kish --> Tehran (90 Pound)
London --> Kish (1000 Pound)
  
Find the cheapest flight between London and Sabezevaar.

rfarhanian replied:
https://github.com/rfarhanian/cs-review/projects/1#card-23313886

rfarhanian committed:
feat(*): implementation
TravellingProblem: LondonSabezevaarShortestPathTest computes the shortest path
TravellingProblem

rfarhanian committed:
feat(pom.xml): pom file was missed
TravellingProblem: LondonSabezevaarShortestPathTest computes the shortest path
TravellingProblem

rfarhanian commented on cs-homework/src/test/java/com/mnaseri/cs/homework/graph/LondonSabzevaarShortestPathTest.java:
```diff
@@ -0,0 +1,91 @@
+            citiesMap.put(cityName, vertex);
+        }
+
+        graph.connect(location("London"), location("Paris")).setDetails(new MutableWeightedEdgeDetails(250));
```
Why are we using Weighted Edge detail?

rfarhanian commented on cs-homework/src/test/java/com/mnaseri/cs/homework/graph/LondonSabzevaarShortestPathTest.java:
```diff
@@ -0,0 +1,91 @@
+    public void testGraph() {
+        DijkstraSingleSourceShortestPathFinder<MutableWeightedEdgeDetails, SimpleVertexDetails> pathFinder = new DijkstraSingleSourceShortestPathFinder<MutableWeightedEdgeDetails, SimpleVertexDetails>();
+        Graph<MutableWeightedEdgeDetails, SimpleVertexDetails> path = pathFinder.find(graph, location("London"));
+        DijkstraSingleSourceShortestPathFinder<MutableWeightedEdgeDetails, SimpleVertexDetails> dijkstra = new DijkstraSingleSourceShortestPathFinder<>();
```
Why Dijkstra? 
What is the complexity of Dijkstra?
What were our alternatives? 
What would be the complexity of the alternatives?

rfarhanian reviewed (commented):
What do you think?
